lib/taskana-core/src/test/java/acceptance/task/GetTaskCommentAccTest.java

```diff
@@ -0,0 +1,104 @@
+package acceptance.task;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import acceptance.AbstractAccTest;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import pro.taskana.common.api.exceptions.NotAuthorizedException;
+import pro.taskana.security.JaasExtension;
+import pro.taskana.security.WithAccessId;
+import pro.taskana.task.api.TaskService;
+import pro.taskana.task.api.exceptions.TaskCommentNotFoundException;
+import pro.taskana.task.api.exceptions.TaskNotFoundException;
+import pro.taskana.task.api.models.TaskComment;
+
+@ExtendWith(JaasExtension.class)
+public class GetTaskCommentAccTest extends AbstractAccTest {
+
+  GetTaskCommentAccTest() {
+    super();
+  }
+
+  @WithAccessId(
+      userName = "user_1_1",
+      groupNames = {"group_1"})
+  @Test
+  void testGetTaskComments()
+      throws TaskCommentNotFoundException, NotAuthorizedException, TaskNotFoundException {
+
+    TaskService taskService = taskanaEngine.getTaskService();
+
+    taskService.getTask("TKI:000000000000000000000000000000000000");
+    List<TaskComment> taskComments =
+        taskService.getTaskComments("TKI:000000000000000000000000000000000000");
+    assertThat(taskComments).hasSize(3);
+  }
+
+  @WithAccessId(
+      userName = "user_1_1",
+      groupNames = {"group_1"})
+  @Test
+  void testGetNonExistingTaskCommentsShouldFail() {
+
+    TaskService taskService = taskanaEngine.getTaskService();
+
+    assertThatThrownBy(
+        () -> taskService.getTaskComments("TKI:000000000000000000000000000000000036"))
+        .isInstanceOf(TaskCommentNotFoundException.class);
+  }
```

Comment:
if the task exists and has no comments, an empty list should be returned. See comment in TaskCommentServiceImpl and TaskService.

Reply:
changed